src/models/Orders.js

```diff
@@ -0,0 +1,105 @@
+const mongoose = require('mongoose');
+
+// mongoose schema for the individual order items
+const orderItemSchema = new mongoose.Schema({
+    // The album field references the Album model and is required
+  album: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Album', // Referencing the Album model
+    required: true,
+  },
+  // The quantity field specifies the number of items in the order
+  quantity: {
+    type: Number,
+    required: true, // Quantity is required
+    min: 1, // Minimum quantity allowed is 1
+  },
+});
+
+// Create the OrderItem model using the defined schema
+const OrderItem = mongoose.model('OrderItem', orderItemSchema);
+
+// Schema for the main Order model
+const orderSchema = new mongoose.Schema({
+    // The user field references the User model and is required
+  user: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'User', // Referencing the User model
+    required: true,
+  },
+  // The orderStatus field indicates the current status of the order
+  orderStatus: {
+    type: String,
+    enum: ['pending', 'payment_successful', 'payment_failed', 'cancelled', 'complete'],
+    required: true, // Order status is required
+    default: 'pending' // Set initial order status as pending
+  },
+  // The tax field represents the tax percentage applied to the order
+  tax: {
+    type: Number,
+    required: true, // Tax percentage is required
+    min: 0, // Tax percentage must be at least 0
+    max: 1, // Tax percentage must be at most 1
+  },
+  // The subtotal field represents the total price before taxes
+  subtotal: {
+    type: Number,
+    required: true,
+  },
+  // The total field represents the total price after taxes
+  total: {
+    type: Number,
+    required: true,
+  },
```

Comment:
Hopefully there wouldn't ever be a case where someone passes in negative numbers to the API; but just in case, we should set some validation here `min: 0`

Reply:
<img width="414" alt="min" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/95896159/e3be6de0-5e8f-4f09-b2e6-c3dd6d86cef5">